packages/background/src/controllers/AccountTrackerController.ts

```diff
@@ -232,12 +232,31 @@ export class AccountTrackerController extends BaseController<AccountTrackerState
                 this._balanceFetchIntervalController.tick(
                     balanceFetchInterval,
                     async () => {
-                        await this.updateAccounts(
-                            {
-                                assetAddresses: [NATIVE_TOKEN_ADDRESS],
-                            },
-                            chainId
+                        // Get addresses from state
+                        const addresses = Object.keys(
+                            this.store.getState().accounts
                         );
+
+                        for (let i = 0; i < addresses.length; i++) {
```

Comment:
isn't this redundant? if `addresses` param is empty, `updateAccounts` method already does this: 

https://github.com/block-wallet/extension/blob/bf89934cec1c3bb4eb7eaa14e40a4032840b3182/packages/background/src/controllers/AccountTrackerController.ts#L712


we can probably include the get tokens method inside too if the `assetAddresses` param is empty to obtain all the addresses for one chain at the same time to prevent executing it on every loop.

also, it'd probably be more clear if we renamed the method to `updateAccountsBalance` or similar

Reply:
yeah but I think we can use the same logic that already exists inside the `updateAccounts` method and check if the `assetAddresses` param comes empty (same as `addresses` param) and get all the tokens for that specific chainId  using a different method like this to prevent the loop again:

```typescript
          const allTokens = []
          Object.values(userTokens).map(t => { 
              const tokens = Object.values(t[chainId]); 
              tokens.map(token => {
                !allTokens.find(t => t.address === token.address) ? allTokens.push(token) : []
              })
          })
          console.log(allTokens)
```

this way we can invoke this method from anywhere knowing that sending the parameters empty will cause this effect (update balance for all accounts and tokens)